### PR TITLE
rewrite to work with config file instead of command line args

### PR DESCRIPTION
**Pull request by phryneas**
This has moved to https://github.com/reduxjs/redux-toolkit/pull/1680

### CODE REVIEW

File: test/__snapshots__/cli.test.ts.snap [hunk 1]
```diff
@@ -1,883 +1,38 @@
...
-
-`;
-
-exports[`CLI options testing should log output to the console when a filename is not specified 1`] = `
-import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query";
-export const api = createApi({
-  baseQuery: fetchBaseQuery({ baseUrl: "/api/v3" }),
-  tagTypes: [],
+exports[`CLI options testing generation with \`config.example.js\` 1`] = `
+import { api } from '../fixtures/emptyApi';
```
Comment (phryneas) [inline test/__snapshots__/cli.test.ts.snap:4]: and we have a first working test. this is what it's gonna be look like :tada:

**Comment by grumpyTofu** [comment]
@phryneas I was able to try this one out. However, I have been running into some issues:

My config:
```ts
import { ConfigFile } from "@rtk-incubator/rtk-query-codegen-openapi/lib";

const {
  parsed: { OPENAPI_DOCS },
} = require("dotenv").config({ path: "./.env.local" });

const config: ConfigFile = {
  schemaFile: OPENAPI_DOCS,
  apiFile: "./src/app/emptyApi.ts",
  //apiImport: "emptyApi",
  outputFile: "./src/app/rtk-api.ts",
  //exportName: "emptyApi",
  hooks: true,
};

export default config;
```
Here is the start of my generated file
```ts
import { api } from "emptyApi";
const injectedRtkApi = api.injectEndpoints({
  endpoints: (build) => ({
    // endpoints
  })
})
```

what I notice is that the import is not resolving correctly. It should be a relative import. If I change it manually it works.
```ts
import { api } from "./emptyApi";
```

I have also noticed that when I use the apiImport option in the config, it changes the import to:
```ts
import { api as emptyApi } from "emptyApi";
const injectedRtkApi = api.injectEndpoints({
  endpoints: (build) => ({
    // endpoints
  })
})
```
but the use of "api" for the injectEnpoints function stays the same, so the whole thing is not working unless you change it to match.

**Comment by phryneas** [comment]
@grumpyTofu good catches! I'll look into those today or tomorrow.

**Comment by phryneas** [comment]
@grumpyTofu I have fixed the bugs you found. But not in this branch, but over in the RTK monorepo: https://github.com/reduxjs/redux-toolkit/pull/1680

Could you give it another try?

**Comment by phryneas** [comment]
This is now published as `@rtk-query/codegen-openapi`.

**Pull request closed without merging**